packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -125,8 +125,13 @@ export function createAppAPI<HostElement>(
   hydrate?: RootHydrateFunction
 ): CreateAppFunction<HostElement> {
   return function createApp(rootComponent, rootProps = null) {
+    if (!isObject(rootComponent)) {
+      __DEV__ && warn(`root component passed to createApp() must be an object.`)
```

Comment:
Something along the lines of

```suggestion
      __DEV__ && warn(`createApp() must receive a root component. You can also pass an empty object:\n  createApp({})`)
```

(needs to be split in multiple lines and formatted